java/ql/src/Telemetry/DatabaseQualityDiagnostics.ql

```diff
@@ -0,0 +1,40 @@
+/**
+ * @name Low Java analysis quality
+ * @description Low Java analysis quality
+ * @kind diagnostic
+ * @id java/diagnostic/database-quality
+ */
+
+import java
+import DatabaseQuality
+
+private newtype TDbQualityDiagnostic =
+  TTheDbQualityDiagnostic() {
+    exists(float percentageGood |
+      CallTargetStatsReport::percentageOfOk(_, percentageGood)
+      or
+      ExprTypeStatsReport::percentageOfOk(_, percentageGood)
+    |
+      percentageGood < 95
+    )
+  }
+
+class DbQualityDiagnostic extends TDbQualityDiagnostic {
+  string toString() {
+    result =
+      "There were significant issues scanning Java code. " +
+        "This may be caused by problems identifying dependencies or use of generated source code, among other reasons -- "
+        + "see other CodeQL diagnostics reported here for more details of possible causes. " +
```

Comment:
```suggestion
        + "see other CodeQL diagnostics reported on the CodeQL status page for more details of possible causes. " +
```
I saw it referred to as "CodeQL status page" in the UI so I think its safe using this language? I'd just avoid saying `reported here` as it is ambiguous - you could also remove `reported here` altogether, but only if you think users will know where to look for further diagnostics!